src/main/scala/ch/epfl/bluebrain/nexus/kg/async/DistributedCache.scala

```diff
@@ -299,25 +325,26 @@ object DistributedCache {
       (replicator ? update).flatMap(handleBooleanUpdate("Timed out while waiting for add project quorum response"))
     }
 
-    private def updateProjectLabel(ref: ProjectRef,
-                                   accountRef: AccountRef,
-                                   proj: Project,
-                                   instant: Instant,
-                                   updateRev: Boolean): Future[Boolean] = {
-      updateProject(ref, proj).withFilter(identity).flatMap { _ =>
-        addProjectToAccount(ref, accountRef, instant, updateRev).withFilter(identity).flatMap { _ =>
-          accountSegment(accountRef).flatMap {
-            case Some(accountLabel) =>
-              val empty = LWWRegister(RevisionedValue[Option[ProjectRef]](0L, None))
-              val value = RevisionedValue[Option[ProjectRef]](proj.rev, Some(ref))
-              val update = Update(projectSegmentKey(ProjectLabel(accountLabel, proj.label)),
-                                  empty,
-                                  WriteMajority(tm.duration))(_.withValue(value))
-              (replicator ? update).flatMap(
-                handleBooleanUpdate("Timed out while waiting for add project quorum response"))
-            case _ => Future.successful(false)
-          }
-        }
+    private def updateProject(ref: ProjectRef,
+                              accountRef: AccountRef,
+                              proj: Project,
+                              instant: Instant,
+                              updateRev: Boolean): Future[Boolean] = {
+      val result = for {
+        _ <- updateProject(ref, proj).withFilter(identity)
```

Comment:
`r <- updateProject(ref, proj) if r` is probably more readable...

Reply:
Then I have to assign variables for each of the for loop elements? I don't find it much easier

Reply:
I agree with Bogdan, `withFilter` in a for comprehension is weird.